src/porepy/utils/array_operations.py

```diff
@@ -264,3 +267,494 @@ def intersect_sets(
     a_in_b[ia] = True
 
     return ia_unique, ib_unique, a_in_b, intersection
+
+
+def unique_rows(data: np.ndarray) -> tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """Finds unique rows in a 2D NumPy array using np.unique with axis=0.
+
+    Parameters:
+        data: A 2D NumPy array of shape `(n, m)`.
+
+    Returns:
+        Tuple containing:
+            An array of the unique rows.
+            Indices of first occurrences of unique rows.
+            Inverse indices mapping original rows to unique rows.
+
+    Example:
+        >>> arr = np.array([[1, 2], [3, 4], [1, 2], [5, 6]])
+        >>> unique_rows(arr)
+        (array([[1, 2], [3, 4], [5, 6]]), array([0, 1, 3]), array([0, 1, 0, 2]))
+
+    """
+    error_message = "Use `unique_vectors_tol` for comparing floats."
+    assert np.issubdtype(data.dtype, np.integer), error_message
+
+    unique, idx, inverse = np.unique(
+        data, axis=0, return_index=True, return_inverse=True
+    )
+    return unique, idx, inverse
+
+
+def unique_columns(
+    data: np.ndarray, preserve_order: bool = True
+) -> tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """Finds unique columns in a 2D numpy array.
+
+    Parameters:
+        data: The 2D numpy array that we want to find unique columns in.
+        preserve_order: If True, the order of the columns will be preserved. If False,
```

Comment:
Please explain in which sense the order is preserved.

Reply:
Explained